hdr_html_canvas_element.md

```diff
@@ -0,0 +1,744 @@
+    float? redPrimaryX;
+    float? redPrimaryY;
+    float? greenPrimaryX;
+    float? greenPrimaryY;
+    float? bluePrimaryX;
+    float? bluePrimaryY;
+    float? whitePointX;
+    float? whitePointY;
+  }
+```
+
+To enable 2D Canvas to use more than 8 bits per pixel.
+
+```idl
+  enum CanvasStorageFormat {
+    "unorm8",
+    "float16", 
+  }
+
+  dictionary CanvasRenderingContext2DSettings {
+    CanvasStorageFormat storageFormat = "unorm8";
+  }
+```
+
+To expose additional color spaces.
+
+```idl
+  partial enum CanvasColorSpace {
+    "srgb-linear",
+    "display-p3-linear",
+    "rec2020-linear",
+  }
+```
+
+To enable ImageData to use more than 8 bits per pixel.
+
+```idl
+  enum ImageDataStorageFormat {
+    "unorm8",
+    "unorm16", 
+    "float32", 
+  }
+
+  typedef (Uint8ClampedArray or Uint16Array or Float32Array) ImageDataArray;
+  partial dictionary ImageDataSettings {
+    ImageDataStorageFormat storageFormat = "unorm8";
+    ImageDataArray data;
+  }
+```
+
+## Example Applications
+
+### Example Set 1: Application With HTML UI on top of a Canvas
+
+In this example set we consider an application in which an HTML UI is used with content in an HTML Canvas.
+
+The beginning of these examples is the same.
+We have a canvas, and a green HTML element on top of it.
+
+```xml
+<html>
+<body>
+<div style='position: relative; width: 500px; height: 500px;'>
+  <canvas id='MyCanvas' width=500px height=500px
+          style='position:absolute; width: 100%; height:100%;'></canvas>
+  <div id='MyGreenUI' style='position:absolute; left:10px; top:10px;
+                             width:100px; height:20px;
+                             background-color:rgb(0, 255, 0);'>
+    MyUIText
+  </div>
+</div>  
+<script>
+  var canvas = document.getElementById('MyCanvas');
+  // ... examples diverge here ...
+</script>
+</body>
+</html>
+```
+
+We can consider two different types of applications in this general context.
+
+#### Example 1A: Application wants the HTML UI to color-match with the HDR Canvas
+
+In this example, the application wants color matching with the HTML UI.
+The most concrete example of this would be a situation where the application uses the HTML color picker element to select colors to use inside the canvas.
+
+In this case, the application will want to use the default mode of ``'extended'``.
+
+```javascript
+    canvas.configureHDR('extended');
+
+    // This green will match the color in MyGreenUI.
+    var context = canvas.getContext('2d', { precision: 'float16' });
+    context.fillStyle = 'rgb(0, 255, 0)';
+    context.fillRect(50, 50, 20, 20);
+```
+
+#### Example 1B: Same as 1A, but using WebGL
+
+This is the same as the above example, but the application is using WebGL.
+
+```javascript
+    canvas.configureHDR('extended');
+
+    // This green will match the color in MyGreenUI.
+    var gl = canvas.getContext('webgl2');
+    gl.configureSwapChain({ format: gl.RGBA16F, colorSpace: 'srgb-linear' });
```

Comment:
The exact API here is still in discussion. It's likely to be:
```
gl.drawingBufferStorage(gl.RGBA16F, canvas.width, canvas.height);
gl.colorSpace = 'srgb-linear';
```

See https://github.com/KhronosGroup/WebGL/pull/3222 .

Reply:
Fixed